examples/iotcc_mqtt.py

```diff
@@ -0,0 +1,170 @@
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+from liota.device_comms.mqtt_device_comms import MqttDeviceComms
+from liota.lib.identity.edge_system_identity import Identity
+from liota.lib.identity.tls_conf import TLSConf
+from liota.lib.transports.mqtt import QoSDetails
+from liota.dccs.iotcc import IotControlCenter
+from liota.dcc_comms.websocket_dcc_comms import WebSocketDccComms
+from liota.entities.edge_systems.dk300_edge_system import Dk300EdgeSystem
+from liota.entities.devices.simulated_device import SimulatedDevice
+from liota.entities.metrics.metric import Metric
+from liota.dccs.dcc import RegistrationFailure
+
+import Queue
+import pint
+
+# getting values from conf file
+config = {}
+execfile('sampleProp.conf', config)
+
+# Create unit registry
+ureg = pint.UnitRegistry()
+
+# Store temperature values in Queue
+kitchen_temperature_data = Queue.Queue()
+living_room_temperature_data = Queue.Queue()
+
+
+# Callback functions
+# To put corresponding values in queue
+def callback_kitchen_temp(client, userdata, message):
+    kitchen_temperature_data.put(float(message.payload))
+
+
+def callback_living_room_temp(client, userdata, message):
+    living_room_temperature_data.put(float(message.payload))
+
+
+# Extract data from Queue
+def get_value(queue):
+    return queue.get(block=True)
+
+# ------------------------------------------------------------------------------------
+# In this example, we demonstrate how data streaming can be done from MQTT channel
+# to IoTCC using LIOTA setting sampling_interval_sec to zero.
+# Here we have two different temperature sensors for two rooms (kitchen and living)
+# Temperature values from sensor will be collected using MQTT channel and redirected
+# to IoTCC data center component
+# ------------------------------------------------------------------------------------
+if __name__ == "__main__":
+
+    # Create DCC object IoTCC using websocket transport
+    # with UID and PASS
+    iotcc = IotControlCenter(config['IotCCUID'], config['IotCCPassword'],
+                             WebSocketDccComms(url=config['WebSocketUrl']))
+
+    try:
+        # Create Edge System identity object with all required certificate details
+        # To connect with a TLS enabled MQTT broker
+        edge_system_identity = Identity(config['cacert'], config['certfile'], config['keyfile'], config['mqtt_username'],
+                               config['mqtt_password'])
+
+        # Encapsulate TLS parameters
```

Comment:
Refactoring change: Move the mqtt channel subscription into a separate method mqtt_subscribe() and call it from the main just for better understanding and clarity that calls are separate from Project ICE.